mitosheet/mitosheet/tests/test_parse.py

```diff
@@ -1134,6 +1134,156 @@ def test_parse(formula, column_header, formula_label, df, python_code, functions
         )
 
 
+# Right now, VLOOKUP is the only formula that allows cross-sheet referencing. In the future, 
+# other cross-sheet references in parser can be added here. 
+VLOOKUP_TESTS = [
+    (
+        '=VLOOKUP(A0, df_2!C:D, 2)',
```

Comment:
Some more parser tests to add:
1. VLOOKUP inside some other function (like a SUM or something else, with other references to the sheet).
2. Some headers in dataframes other than just strings (e.g. some numbers or booleans in the cross sheet reference.


The other major set of tests we should add: set column formula tests!

We have the ability to test the entire setting formula loop from start to finish. Notably, this sort of backend testing would mean you get to spend less time manually writing formulas in the frontend -- this is the "lead with tests" idea we've talked about before.

I'll let you write these tests yourself - many of the parser tests likely will transport over, but also there are some new, more realistic scenarios that might be nice to test!